release-lib-master.Jenkinsfile

```diff
@@ -0,0 +1,176 @@
+            name: volume-0
+          - mountPath: /home/jenkins/gnupg_keys
+            name: volume-1
+          - mountPath: /mnt/ocp
+            name: volume-2
+          workingDir: /home/jenkins
+        volumes:
+        - name: volume-0
+          secret:
+            defaultMode: 420
+            secretName: sonatype-secrets
+        - name: volume-1
+          secret:
+            defaultMode: 420
+            secretName: gnupg
+        - name: volume-2
+          configMap:
+            defaultMode: 420
+            name: jenkins-openshift-mappings
+      """
+    }
+  }
+  options {
+    //timestamps()
+    timeout(time: 120, unit: 'MINUTES')
+  }
+  environment {
+    PIPELINE_NAMESPACE = readFile('/run/secrets/kubernetes.io/serviceaccount/namespace').trim()
+    PIPELINE_USERNAME = sh(returnStdout: true, script: 'id -un').trim()
+  }
+  stages {
+    stage('git checkout') {
+      steps{
+        script{
+          checkout([$class      : 'GitSCM', branches: [[name: params.LIB_GIT_BRANCH]], doGenerateSubmoduleConfigurations: false,
+                    extensions  : [[$class: 'RelativeTargetDirectory', relativeTargetDir: params.LIB_NAME]], submoduleCfg: [],
+                    userRemoteConfigs: [[url: params.LIB_GIT_REPO]]])
+
+          echo "Prepare the release of ${params.LIB_GIT_REPO} branch: ${params.LIB_GIT_BRANCH}"
+
+          sh """#!/bin/bash
+          cd ${params.LIB_NAME}
+          git checkout ${params.LIB_GIT_BRANCH}
+          gpg --allow-secret-key-import --import /home/jenkins/gnupg_keys/private_key.txt
+          """
+        }
+      }
+    }
+    stage('Setup and Formating'){
+      steps{
+        script{
+          withCredentials([
+            usernamePassword(credentialsId:'GitHub-Bot', passwordVariable:'BOT_PASSWORD', usernameVariable:'BOT_USERNAME'),
+            usernamePassword(credentialsId:'OSS-Nexus-Bot', passwordVariable:'OSS_BOT_PASSWORD', usernameVariable:'OSS_BOT_USERNAME'),
+            string(credentialsId: 'gnupg_passphrase', variable: 'PASSPHRASE')
+          ]) {
+            dir(params.LIB_NAME){
+              sh """
+              mkdir -p /home/jenkins/.m2
+              cp ../settings-release.xml /home/jenkins/.m2/settings.xml
+              sed -i 's/{{_PASSPHRASE}}/${PASSPHRASE}/g' /home/jenkins/.m2/settings.xml
+              sed -i 's/{{_USERNAME}}/${OSS_BOT_USERNAME}/g' /home/jenkins/.m2/settings.xml
+              sed -i 's/{{_PASSWORD}}/${OSS_BOT_PASSWORD}/g' /home/jenkins/.m2/settings.xml
+              sed -i s,git@github.com:Commonjava/indy.git,https://`python3 -c 'print("${params.LIB_GIT_REPO}".split("//")[1])'`,g pom.xml
+              sed -i s,https://github.com/Commonjava/indy.git,https://`python3 -c 'print("${params.LIB_GIT_REPO}".split("//")[1])'`,g pom.xml
+              git config --global user.email "${params.BOT_EMAIL}"
+              git config --global user.name "${BOT_USERNAME}"
+              mvn -B -s ../settings.xml -Pformatting clean install
```

Comment:
I think we need to include the `cq` profile here, but we might want to just call the `license:format` goal directly instead of the `install` lifecycle phase. It would avoid problems with things like PMD that might be enabled by that profile.